src/components/bookmark-button/helpers/with-dialog.tsx

```diff
@@ -1,66 +1,65 @@
 import { useCallback, useState } from 'react'
-import Dialog from 'components/dialog'
+import { Tutorial } from 'lib/learn-client/types'
 import useAuthentication from 'hooks/use-authentication'
+import { useBookmarkMutations, useIsBookmarked } from 'hooks/bookmarks'
+import Dialog from 'components/dialog'
 import { BookmarkButtonProps } from '../types'
 import BookmarkSignInPrompt from '../sign-in-dialog'
+import makeBookmarkToast from '../toast/make-bookmark-toast'
 
 /**
  * This HOC serves as a 'controller'
  * for rendering the various state of bookmark
  * interaction. It accepts a UI component to render,
- * and renders a dialog if unauthenticed
- *
- * Eventually this could be extended to
- * post / delete based on interactions and
- * trigger toast notifications based on the result.
+ * and renders a dialog if unauthenticated.
  */
 
 export function withDialog(BookmarkComponent: React.FC<BookmarkButtonProps>) {
 	return function BookmarkComponentWithDialog({
-		isBookmarked,
-	}: Pick<BookmarkButtonProps, 'isBookmarked'>) {
+		tutorialId,
+	}: {
+		tutorialId: Tutorial['id']
+	}) {
 		const { isAuthenticated, signIn } = useAuthentication()
+		const { isBookmarked } = useIsBookmarked({ tutorialId })
+		const { addBookmark, removeBookmark } = useBookmarkMutations()
```

Comment:
So I think we should lift this up to `ConnectedTutorialCard` and that can serve as the 'controller'. That controller could also pass down the `addOrRemoveBookmark` handler. And we would only have to handle the `tutorialId` at that level instead of passing it down. Thoughts?

Reply:
If we take that approach, this component will have the sole responsibility of _render a dialog or not_. And the `Connected` component will have the responsibility of _fetch the dynamic data and decide what to render in the eyebrow_.